usr/src/lib/libzfs/common/libzfs_sendrecv.c

```diff
@@ -609,11 +621,23 @@ send_iterate_snap(zfs_handle_t *zhp, void *arg)
 {
 	send_data_t *sd = arg;
 	uint64_t guid = zhp->zfs_dmustats.dds_guid;
+	uint64_t txg = zhp->zfs_dmustats.dds_creation_txg;
 	char *snapname;
 	nvlist_t *nv;
 
 	snapname = strrchr(zhp->zfs_name, '@')+1;
 
+	if (sd->parent_tosnap_txg != 0 && txg > sd->parent_tosnap_txg) {
```

Comment:
Can you explain the motivation behind this change?  It seems orthogonal to the problem described in the bug report.  In this case the tosnap does exist in every filesystem, but they were created at different times (later in the child than its parent).  This previously worked, why do you want to make it silently fail now?

Reply:
I know this is all quite old now, but I only just found this thread and I need to point out that the following statement is not correct:

> _**But as before**_, new algorithm returns an error in the case where the nested dataset A/B does not contain a snapshot tosnap AND dataset A/B was created earlier than the snapshot tosnap for its parent dataset A.

See https://github.com/zfsonlinux/zfs/issues/5196#issuecomment-520311993